lib/rules/selector-not-notation/index.js

```diff
@@ -130,6 +130,19 @@ const rule = (primary, _, context) => {
 	};
 };
 
+/**
+ * @param {Node[]} nodes
+ */
+const getLastConsecutiveNot = (nodes) => {
+	let i = 0;
+
+	for (; i < nodes.length; i++) {
+		if (!isNot(/** @type {Node} */ (nodes[i]))) break;
+	}
+
+	return nodes[i - 1];
```

Comment:
[suggest] What about making the code readable by `findIndex()` like this? IMHO, an old-style loop using an index is not so readable.

In addition, what about returning `Node` instead of `Node | undefined` by moving `assert()`?

```suggestion
 * @param {Node[]} nodes
 * @returns {Node}
 */
const getLastConsecutiveNot = (nodes) => {
	const index = nodes.findIndex((node) => !isNot(node));
	const found = nodes[index - 1] || nodes[0];

	assert(found);

	return found;
```

Reply:
This is obviously more readable and elegant.
Thanks!

Reply:
This is obviously more readable and elegant, thanks!
I see a problem with `const found = nodes[index - 1] || nodes[0];` though.
Ill add a test if it's not already covered.